lstests/core/service.py

```diff
@@ -0,0 +1,1517 @@
+        self.hide_all = cla.hide_all
+
+    @staticmethod
+    def parse():
+        """Parse for command-line arguments for Logger, Service, Cluster.
+
+        Appending "-h/--help" to any test script will show help information."""
+        desc = color.decorate("Launcher Service-based EOSIO Testing Framework", style="underline", fcolor="green")
+        left = 5
+        form = lambda text, value=None: "{} ({})".format(helper.pad(text, left=left, total=55), value)
+        parser = argparse.ArgumentParser(description=desc, add_help=False,
+                 formatter_class=lambda prog: argparse.RawTextHelpFormatter(prog, max_help_position=50))
+        parser.add_argument("-h", "--help", action="help", help=" " * left + HELP_HELP)
+        # service-related options
+        parser.add_argument("-a", "--addr", type=str, metavar="IP", help=form(HELP_ADDR, DEFAULT_ADDR))
+        parser.add_argument("-o", "--port", type=int, help=form(HELP_PORT, DEFAULT_PORT))
+        parser.add_argument("-w", "--wdir", type=str, metavar="PATH", help=form(HELP_WDIR, DEFAULT_WDIR))
+        parser.add_argument("-f", "--file", type=str, metavar="PATH", help=form(HELP_FILE, DEFAULT_FILE))
+        parser.add_argument("-g", "--gene", type=str, metavar="PATH", help=form(HELP_GENE, DEFAULT_GENE))
+        parser.add_argument("-s", "--start", action="store_true", default=None, help=form(HELP_START, DEFAULT_START))
+        parser.add_argument("-k", "--kill", action="store_true", default=None, help=form(HELP_KILL, DEFAULT_KILL))
+        # cluster-related options
+        parser.add_argument("-c", "--cdir", metavar="PATH",
+                            help=form(HELP_CDIR, DEFAULT_CDIR))
+        parser.add_argument("-i", "--cluster-id", dest="cluster_id", type=int, metavar="ID",
+                            help=form(HELP_CLUSTER_ID, DEFAULT_CLUSTER_ID))
+        parser.add_argument("-n", "--node-count", type=int, metavar="NUM",
+                            help=form(HELP_NODE_COUNT, DEFAULT_NODE_COUNT))
+        parser.add_argument("-p", "--pnode-count", type=int, metavar="NUM",
+                            help=form(HELP_PNODE_COUNT, DEFAULT_PNODE_COUNT))
+        parser.add_argument("-q", "--producer-count", type=int, metavar="NUM",
+                            help=form(HELP_PRODUDCER_COUNT, DEFAULT_PRODUDCER_COUNT))
+        parser.add_argument("-u", "--unstarted-count", type=int, metavar="NUM",
+                            help=form(HELP_UNSTARTED_COUNT, DEFAULT_UNSTARTED_COUNT))
+        parser.add_argument("-t", "--topology", type=str, metavar="SHAPE", help=form(HELP_TOPOLOGY, DEFAULT_TOPOLOGY),
+                            choices={"mesh", "bridge", "line", "ring", "star", "tree"})
+        parser.add_argument("-x", "--center-node-id", type=int, metavar="ID",
+                            help=form(HELP_CENTER_NODE_ID, DEFAULT_CENTER_NODE_ID))
+        parser.add_argument("-dnewa", "--dont-newacco", action="store_true", default=None,
```

Comment:
Thanks for the comment. That `dont` is used instead of `no` is primarily because it's also widely used in the original Python tests (in `eos/tests`) like `dontLaunch`/`dont_launch`/`dont-launch`, `dontBootstrap` and `dontKill`. More than that, it's also used in `cleos` (as in `--dont-broadcast`), which is a major program of `eos`. This attempt is for consistency. Again, it's more or less subjective.

As to the multi-letter options with a single dash prefix, I do plan to have a final review on the command-line option letters after everything else is settled, as by then we shall know what features are really needed. Some features may be introduced/dropped during the code review. Thanks for this.

Reply:
@b1aafulei I'm going to push back on keeping any multi-letter options starting with a single dash; that includes the ones starting with `h` as well.

Reply:
@arhag Thanks for the suggestion. All multi-letter options starting with `-h` have been removed. In general, in case of any fine-tune on the command-line options, I will keep this as a principle: there will be no multi-letter command-line options that start with a single dash.